hack/test-stirunimage.sh

```diff
@@ -136,6 +136,12 @@ docker run test >& "${WORK_DIR}/s2i-override-run.log"
 grep "Running custom run" "${WORK_DIR}/s2i-override-run.log"
 check_result $? "${WORK_DIR}/s2i-override-run.log"
 
+test_debug "s2i build with add-host option"
```

Comment:
I think that this should be two separate tests, less one give the wrong result and they both pass.
Also, it should give some kind of specific error if rubygems.org can not be found.

Reply:
You could set rubygems.org:127.0.0.1 and it should fail (since localhost is not a rubygems server) and you should get an error message about rubygems.org not being found

Reply:
i have change to 0.0.0.0 because error of 172.0.0.1 is much more unclear and it takes more time because it is the LOOPBACK internet interface. I also squash all my commits.  @bparees